readme.html

```diff
@@ -72,53 +78,236 @@ <h2>Installation</h2>
     This directory is configured to be deleted, after Openfire quits. Some Operating Systems are known to periodically
     purge these temporary directories, which is another potential source of data loss. This plugin can be configured to
-    use any other directory for data storage, by providing the path for that directory in the property <samp>plugin.httpfileupload.fileRepo</samp>.
+    use any other directory for data storage, by providing the path for that directory in either property listed above.
     Note that the directory is expected to exist (this plugin will not attempt to create it if it is missing). Be sure
     that the process owner of the Openfire process has enough permissions to read and write in that directory.
 </p>
 
+<h3>Maximum file size</h3>
+
+<table>
+    <tr><th>XML property</th> <td><samp>plugin.httpfileupload.serverspecific.maxFileSize</samp></td></tr>
+    <tr><th>Property</th> <td><samp>plugin.httpfileupload.maxFileSize</samp></td></tr>
+</table>
+
+<p>
+    To configure the maximum allowable file size to be uploaded by clients, the property defined above can be set to a
+    value in bytes. If not set, a default value is used. To disable the file size limitation, set a property to a value
+    of <samp>-1</samp>.
+</p>
+
+<h3>Announced web endpoint</h3>
+
 <p>
-    To configure the maximum allowable file size to be uploaded by clients, the property
-    <samp>plugin.httpfileupload.maxFileSize</samp> can be set to a value in bytes. If not set, a default value is used. To
-    disable the file size limitation, set this property to a value of <samp>-1</samp>.
+    The plugin will send URLs to the client that are to be used to upload or download data. To following properties can
+    be used to control the composition of the URLs that are generated.
 </p>
 
 <p>
-    To change the HTTP server protocol from the default value of "https" to "http" for example, the property <samp>plugin.httpfileupload.announcedWebProtocol</samp> can be set to new value.
+    To change the HTTP server protocol from the default value of "https" to "http" for example, use these properties:
 </p>
 
+<table>
+    <tr><th>XML property</th> <td><samp>plugin.httpfileupload.serverspecific.announcedWebProtocol</samp></td></tr>
+    <tr><th>Property</th> <td><samp>plugin.httpfileupload.announcedWebProtocol</samp></td></tr>
+</table>
+
 <p>
-    To change the HTTP server hostname from the default value used by Openfire, the property <samp>plugin.httpfileupload.announcedWebHost</samp> can be set to new value.
+    To change the HTTP server hostname from the default value used by Openfire, use these properties:
 </p>
 
+<table>
+    <tr><th>XML property</th> <td><samp>plugin.httpfileupload.serverspecific.announcedWebHost</samp></td></tr>
+    <tr><th>Property</th> <td><samp>plugin.httpfileupload.announcedWebHost</samp></td></tr>
```

Comment:
Should this be System Property, to be super-specifc?
```suggestion
    <tr><th>System Property</th> <td><samp>plugin.httpfileupload.announcedWebHost</samp></td></tr>
```